R/spread-process.R

```diff
@@ -165,32 +172,34 @@ setMethod("spread",
                                             pairs=FALSE))
               }
               #browser()
-
-
+              
+              
               #if there is only one potential, R converts this to a vector, instead of a matrix.
               # Force it back to a matrix
-#              if(length(potentials)==2) {
-#                potentials <- matrix(potentials,ncol=2)
-#              }
-
+              #              if(length(potentials)==2) {
+              #                potentials <- matrix(potentials,ncol=2)
+              #              }
+              
               # drop those ineligible
               #              if (!is.null(mask))
               #                potentials <- matrix(potentials[potentials[,2] %in% masked,], ncol=2)
-
+              
               # only accept cells that have no fire yet
               # Need to call matrix because of the cast where there is only one cell
               #potentials <- matrix(potentials[spreads[potentials[,2]]==0,], ncol=2)
               potentials <- potentials[spreads[potentials[,2]]==0,,drop=FALSE]
-
+              
               # If one pixels is selected as potential by more than one source
               #  Remove the duplication, and reorder the potentials so that it is not
               #  always the "first one", i.e., closest to top left of map, that is kept.
-              if(nrow(potentials)>0) {
-                potentials <- potentials[sample.int(nrow(potentials)),,drop=FALSE]
-              }
-              potentials <- potentials[!duplicated(potentials[,2]),,drop=FALSE]
-
-
+              if(NROW(potentials)>0) {
+                if(anyDuplicated(potentials[,2])){
+                  ## For testing purposes (to benchmark)
```

Comment:
I am not sure that this "removal of duplicates" is correct, as a cell may
be under attack by more than one neighbour at the same time....this would
tend to reduce the rate of spread, maybe by quite a lot.

sc

On Tue, Apr 7, 2015 at 1:48 PM, J7git <notifications@github.com> wrote:

> In R/spread-process.R
> <https://github.com/PredictiveEcology/SpaDES/pull/148#discussion_r27902092>
> :
>
> >                # If one pixels is selected as potential by more than one source
> >                #  Remove the duplication, and reorder the potentials so that it is not
> >                #  always the "first one", i.e., closest to top left of map, that is kept.
> > -              if(nrow(potentials)>0) {
> > -                potentials <- potentials[sample.int(nrow(potentials)),,drop=FALSE]
> > -              }
> > -              potentials <- potentials[!duplicated(potentials[,2]),,drop=FALSE]
> > -
> > -
> > +              if(NROW(potentials)>0) {
> > +                if(anyDuplicated(potentials[,2])){
> > +                  ## For testing purposes (to benchmark)
>
> It's necessary code you are right. I need to benchmark that part to see if
> it's worth doing a check every time to avoid a sample.int() or let the
> code as it was. I need to find the time to do it.
>
> The change is made so that we will keep it if the speed actually improve
> overall.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PredictiveEcology/SpaDES/pull/148/files#r27902092>.
>



-- 
Steve Cumming
Canada Research Chair in Boreal Ecosystems Modelling
Département des sciences du bois et de la forêt
Université Laval
(418) 656-2131 poste 2593